sysvinit/gentoo/cloud-init-hotplug

```diff
@@ -0,0 +1,18 @@
+#!/sbin/openrc-run
+
+command="/usr/lib/cloud-init/cloud-init-hotplugd"
+command_background=yes
+pidfile="/run/$RC_SVCNAME.pid"
+
+depend() {
+  before cloud-init
+}
+
+start_pre() {
+   if grep -q 'cloud-init=disabled' /proc/cmdline; then
+     ewarn "$RC_SVCNAME is disabled via /proc/cmdline."
+   elif test -e /etc/cloud/cloud-init.disabled; then
+     ewarn "$RC_SVCNAME is disabled via cloud-init.disabled file"
+   fi
+   eend 0
```

Comment:
I don't see us invoking $command anywhere as I've seen in other openrc scripts. Is this done automatically by s6-svcscan for us automatically? I'm also uncertain how we really prevent this service from running if s6-svcscan is automatically calling start on the $command. I'd expect we have some facility to disable trying to start a service in openrc due to unmet dependencies/config conditions.

Reply:
Otherwise if s6-svcscan isn't our service supervisor, don't we need a start/stop/restart function in this openrc file that would only attempt to start the service if our warning conditions weren't triggered?

Reply:
> Is this done automatically by s6-svcscan for us automatically?

I'm not sure why you are focusing on s6-svcscan.

Whilst s6 can be used as a supervisor for OpenRC it is *not* the default on - stop-stop-daemon is. In order to use s6-svcscan (which this script is not doing) you would need to specifically enable it, see: https://github.com/OpenRC/openrc/blob/master/s6-guide.md

Reply:
Yes s6-svscan was just bad google-fu and searching on my part. Thanks for the link.

My question then becomes, this particular openrc script doesn't appear to actually provide any start/stop bit of functionality or invoking any commands. It only tests and emits warning messages about disabled conditions and exits 0. The other gentoo sysvinit openrc scripts all seem to invoke `cloud-init init ...` on the command line when cloud-init is not in a disabled state. Does this current script actually need to invoke `/usr/lib/cloud-init/cloud-init-hotplugd` directly when no disabled/warning conditions exist?

I'm wondering if the openrc script needs to be something like the following so we can be sure that hotplugd service is only started when and if cloud-init is not disabled?

```
start() {
start() {
  if grep -q 'cloud-init=disabled' /proc/cmdline; then
    ewarn "$RC_SVCNAME is disabled via /proc/cmdline."
  elif test -e /etc/cloud/cloud-init.disabled; then
    ewarn "$RC_SVCNAME is disabled via cloud-init.disabled file"
  else
    # only get here if cloud-init is enabled
    start-stop-daemon --start --exec $command --pidfile $pidfile
  fi
  eend 0
}
```

Reply:
> My question then becomes, this particular openrc script doesn't appear to actually provide any start/stop bit of functionality or invoking any commands.

That's the magic of OpenRC - by default you are **not** expected to write start() and stop() functions, OpenRC uses builtin functions which use the "command", "pidfile", and other variables to know what to start and stop. From OpenRC [docs](https://github.com/OpenRC/openrc/blob/master/service-script-guide.md):

```
All scripts must have start/stop/status functions, but defaults are provided and should be used unless you have a very strong reason not to use them.
```

However in this case I've now added a start() function as I couldn't do what I wanted to do via a start_pre() function.